run/websockets/public/index.js

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Material Design functionality
+mdc.textField.MDCTextField.attachTo(document.querySelector('.name'));
+mdc.textField.MDCTextField.attachTo(document.querySelector('.room'));
+mdc.ripple.MDCRipple.attachTo(document.querySelector('.signin'));
+mdc.ripple.MDCRipple.attachTo(document.querySelector('.send'));
+
+// Hide Chatroom on start
+$(document).ready(() => {
+  $('#chatroom').hide();
+});
+
+// [START cloudrun_websockets_initialization]
+// Initialize Socket.io
+const socket = io('', {
+  transports: ['websocket'],
+});
+// [END cloudrun_websockets_initialization]
+
+let name;
+let room;
```

Comment:
question: Why initialize these variables here? It seems you don't need them in global scope.
suggestion: if not needed here, add the let keyword to the variable assignments at line 38.

Reply:
This is needed for sending the 'updateSocketId' message. I will add a comment.